library/core/src/ptr/const_ptr.rs

```diff
@@ -656,6 +656,38 @@ impl<T: ?Sized> *const T {
         self.wrapping_offset((count as isize).wrapping_neg())
     }
 
+    /// Sets the pointer value to `ptr`.
+    ///
+    /// In case `self` is a (fat) pointer to an unsized type, this operation
+    /// will only affect the pointer part, whereas for (thin) pointers to
+    /// sized types, this has the same effect as a simple assignment.
```

Comment:
This should discuss pointer provenance. Will the resulting pointer have provenance of `self` or of `val`? I think it will be quite a bit less useful to have that of `self` as the usecase of pivoting a pointer to a copy of the bytes (i.e. 'moving' an unsized value) couldn't work otherwise.

Reply:
I agree on both counts: provenance needs to be mentioned in some fashion by the docs, and those semantics (takes provenance from val, not self) are what makes sense.

Basically, fat_ptr.set_ptr_value(new_ptr) should have the semantics of create_fat_pointer(new_ptr, ptr_metadata_of(fat_ptr)).

I don't think there's a way in which "physical bits of val but provenance of self" makes sense, but it's still good to be clear that provenance of the new value is kept alongside its target.

Reply:
The provenance will be that of `val`. There is currently no operation in Rust that can take the provenance of one pointer and combine it with the value (the memory location it points to) of another pointer.